### PR TITLE
[8.0][FIX] business requirement deliverable cost bug regards on change decorator could not be retrieved properly pricelist

### PR DESCRIPTION
**Pull request by martintorresvm**

### CODE REVIEW

**Comment by seb-elico** [comment]
@victormartinelicocorp LGTM but I'm wondering why there's some code in common with #62 , is it on purpose?

**Comment by martintorresvm** [comment]
> @victormartinelicocorp LGTM but I'm wondering why there's some code in common with #62 , is it on purpose?

@seb-elico I just wanted to take the benefits of the fix on the unmerged #62. At the end the issue solved by that PR is blocking IMO to test properly the functionality. And not really sure about how do it right with Git commands

**Comment by elicoidal** [comment]
@victormartinelicocorp Can you fix travis?

**Comment by elicoidal** [comment]
:+1:

**Comment by martintorresvm** [comment]
> @victormartinelicocorp Can you fix travis?

@elicoidal  done. But I need to include UT (fix Coverage/Coveralls) to check the new possible paths.

File: business_requirement_deliverable_cost/models/business.py [hunk 1]
```diff
@@ -102,7 +103,7 @@ def product_id_change(self):
                     pricelist=pricelist_id.id,
                     uom=self.uom_id.id,
                 )
-                sale_price_unit = product.list_price
+                sale_price_unit = product.price
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py]: Looks strange to me. Can you elaborate this change?

File: business_requirement_deliverable_cost/models/business.py [hunk 2]
```diff
@@ -144,10 +136,10 @@ def product_uom_change(self):
                     uom=self.uom_id.id,
                 )
                 unit_price = product.standard_price
-                sale_price_unit = product.list_price
+                sale_price_unit = product.price
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py]: idem

**Review by elicoidal — COMMENTED** [review]
some explanations needed

File: business_requirement_deliverable_cost/models/business.py (hunk 1 above)
Comment (martintorresvm) [reply business_requirement_deliverable_cost/models/business.py]: `list_price` is a base price set by user and `price` is the computed field `Product price`. When context changes is re-calculated. Weird to the user check that if change `qty` or `uom_id` parameter will back to the base price without take on count the appropriate pricelist or unit of measure.

File: business_requirement_deliverable_cost/models/business.py (hunk 2 above)
Comment (martintorresvm) [reply business_requirement_deliverable_cost/models/business.py]: same as above

File: business_requirement_deliverable_cost/models/business.py [hunk 3]
```diff
@@ -139,7 +161,7 @@ def action_button_update_estimation(self):
                 for resource in deliverable.resource_ids:
                     pricelist_id = resource._get_pricelist()
                     partner_id = resource._get_partner()
-                    resource.sale_price_unit = resource.product_id.lst_price
+                    resource.sale_price_unit = resource.product_id.list_price
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py]: @victormartinelicocorp just checking: this one is OK?

**Comment by elicoidal** [comment]
One check/question. Other than that :+1:

**Comment by elicoidal** [comment]
By the way, we need to fix travis

File: business_requirement_deliverable_cost/models/business.py (hunk 3 above)
Comment (martintorresvm) [reply business_requirement_deliverable_cost/models/business.py]: Yes it's ok. I will explain. `lst_price` is a related field to `list_price` so in the end is the same value but I have checked on the others BR modules and regards product we use `list_price` one so I rather use the same on all the modules.

`'list_price': fields.float('Sale Price', digits_compute=dp.get_precision('Product Price'), help="Base price to compute the customer price. Sometimes called the catalog price."),`

`'lst_price' : fields.related('list_price', type="float", string='Public Price', digits_compute=dp.get_precision('Product Price')),`

**Comment by martintorresvm** [comment]
@elicoidal @seb-elico All test has been passed.

File: business_requirement_deliverable_cost/models/business.py (hunk 3 above)
Comment (elicoidal) [reply business_requirement_deliverable_cost/models/business.py]: @victormartinelicocorp My point is more on list_price and price

**Comment by elicoidal** [comment]
@astirpe @mmalorni can you have a review?

File: business_requirement_deliverable_cost/models/business.py (hunk 3 above)
Comment (martintorresvm) [reply business_requirement_deliverable_cost/models/business.py]: @elicoidal  I am sorry but not sure what you mean with `list_price` and `price`. Maybe you are asking why I don't use `price` on this situation?

Comment (elicoidal) [reply business_requirement_deliverable_cost/models/business.py]: @victormartinelicocorp yes

**Review by elicoidal — COMMENTED** [review]
Can you resolve conflicts?

**Comment by martintorresvm** [comment]
> Can you resolve conflicts?

@elicoidal Done. But need to fix Travis now

**Comment by elicoidal** [comment]
> Done. But need to fix Travis now

@victormartinelicocorp any news?

**Comment by martintorresvm** [comment]
> @victormartinelicocorp any news?

@elicoidal Sorry but recently busy with others projects. I'll try to save some time today to work on it if I'll keep you post.

**Comment by martintorresvm** [comment]
@elicoidal Here we have another kind of problem to fix remain small issues with onchange methods for Resource Lines.

As you know we'd like to link calculate revenue with the pricelist from BR 1 master project or 2 customer or if not with default price of product. 

onchange() method allows you to access from DL to BR or RL to DL. If try to access like RL -> DL -> BR is not possible to check pricelist information accurate.

Options:
1) If someone change UoM just re-calculate with the new unit of measure.
2) If really someone want to recalculate based on actual pricelist leave this logic only for the Button "Update resource lines pricelist"

**Comment by elicoidal** [comment]
As discussed, please check whether you can store in the intermediate model the pricelist ids.
Otherwise, another 2 solution:
* RL depends on BR instead of DL (but that is a main change in modelization)
* We assume UoM change doesnot change the price (normally this is only a factor).

**Comment by martintorresvm** [comment]
@elicoidal I rather close this PR until Issue #82 is accepted and done otherwise never could be done calculate properly on_change methods on RL.

**Comment by elicoidal** [comment]
:+1:

**Pull request closed without merging**